File: outdoor-saas-be/src/main/java/com/touhuwai/service/BarrierGateService.java
package com.touhuwai.service;

import com.touhuwai.common.PageResult;
import com.touhuwai.entity.BarrierGate;

import java.util.List;

/**
 * 道闸设备服务接口
 */
public interface BarrierGateService {
    
    /**
     * 根据ID查询道闸
     * @param id 道闸ID
     * @return 道闸信息
     */
    BarrierGate getById(Integer id);
    
    /**
     * 根据道闸编号查询
     * @param gateNo 道闸编号
     * @return 道闸信息
     */
    BarrierGate getByGateNo(String gateNo);
    
    /**
     * 查询所有道闸
     * @return 道闸列表
     */
    List<BarrierGate> getAll();
    
    /**
     * 分页查询道闸列表
     * @param barrierGate 查询条件
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    PageResult<BarrierGate> getPage(BarrierGate barrierGate, Integer pageNum, Integer pageSize);
    
    /**
     * 根据社区ID查询道闸列表
     * @param communityId 社区ID
     * @return 道闸列表
     */
    List<BarrierGate> getByCommunityId(Integer communityId);
    
    /**
     * 新增道闸
     * @param barrierGate 道闸信息
     * @return 影响的行数
     */
    int add(BarrierGate barrierGate);
    
    /**
     * 批量新增道闸
     * @param list 道闸列表
     * @return 影响的行数
     */
    int batchAdd(List<BarrierGate> list);
    
    /**
     * 更新道闸
     * @param barrierGate 道闸信息
     * @return 影响的行数
     */
    int update(BarrierGate barrierGate);
    
    /**
     * 根据ID删除道闸
     * @param id 道闸ID
     * @return 影响的行数
     */
    int delete(Integer id);
    
    /**
     * 批量删除道闸
     * @param ids ID列表
     * @return 影响的行数
     */
    int batchDelete(List<Integer> ids);
}

File: outdoor-saas-fe/src/services/barrierGateService.ts
/**
 * 道闸设备管理服务
 */
import request from './api';
import type { BarrierGateQueryParam, PageResult } from '../types/query';

export interface BarrierGate {
  id: number;
  gateNo: string;
  deviceNo: string;
  doorLocation: string;
  devicePosition?: number;
  screenPosition?: number;
  lightboxDirection?: number;
  communityId: number;
  communityName?: string;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
  createdAt?: string;
  updatedAt?: string;
}

export const barrierGateService = {
  // 获取所有道闸
  getAll: () => request<BarrierGate[]>('/barrier-gate/list'),

  // 根据ID获取道闸
  getById: (id: number) => request<BarrierGate>(`/barrier-gate/${id}`),

  // 根据道闸编号获取
  getByNo: (gateNo: string) => request<BarrierGate>(`/barrier-gate/no/${gateNo}`),

  // 分页查询道闸
  getPage: (page: number = 1, size: number = 10) =>
    request<BarrierGate[]>('/barrier-gate/page', {
      method: 'POST',
      body: JSON.stringify({ page, size }),
    }),

  // 根据社区ID获取道闸
  getByCommunity: (communityId: number) =>
    request<BarrierGate[]>(`/barrier-gate/community/${communityId}`),

  // 创建道闸
  create: (data: Partial<BarrierGate>) =>
    request<BarrierGate>('/barrier-gate', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 更新道闸
  update: (data: Partial<BarrierGate>) =>
    request<BarrierGate>('/barrier-gate', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  // 删除道闸
  delete: (id: number) =>
    request<void>(`/barrier-gate/${id}`, { method: 'DELETE' }),

  // 批量删除道闸
  deleteBatch: (ids: number[]) =>
    request<void>('/barrier-gate/batch', {
      method: 'DELETE',
      body: JSON.stringify(ids),
    }),

  // 过滤查询道闸列表（不分页）
  filter: (params: BarrierGateQueryParam) =>
    request<PageResult<BarrierGate>>('/barrier-gate/filter', {
      method: 'POST',
      body: JSON.stringify(params),
    }),

  // 过滤查询道闸列表（分页）
  filterPage: (params: BarrierGateQueryParam) =>
    request<PageResult<BarrierGate>>('/barrier-gate/filter/page', {
      method: 'POST',
      body: JSON.stringify(params),
    }),
};

export default barrierGateService;
